compiler/src/Language/Bond/Codegen/Java/Enum_java.hs

```diff
@@ -30,37 +30,73 @@ package #{javaPackage};
 
     typeDefinition Enum {..} = [lt|
 #{Java.generatedClassAnnotations}
-public class #{declName} {
-    #{newlineSep 1 constant enumConstantsWithInt}
+public final class #{declName} {
+
+    public static final class Values {
+        private Values() {}
+
+        #{newlineSep 2 constantIntValueDecl enumConstantsWithInt}
+    }
+
+    #{newlineSep 1 constantObjectDecl enumConstantsWithInt}
 
     public final int value;
 
     public #{declName}(int value) { this.value = value; }
 
     @Override
-    public boolean equals(Object other) {
-        if (!(other instanceof #{declName})) { return false; }
-        return this.value == ((#{declName}) other).value;
-    }
+    public boolean equals(Object other) { return (other instanceof #{declName}) && (this.value == ((#{declName}) other).value); }
+ 
+    @Override
+    public int hashCode() { return this.value; }
 
     @Override
-    public int hashCode() {
-        return this.value;
+    public String toString() { return String.valueOf(this.value); }
```

Comment:
I would expect `Color.RED.toString()` to return `"RED"`, not `0`. This is what true Java enums do.

You'll need a new member for this, but I think it's in scope for this change.